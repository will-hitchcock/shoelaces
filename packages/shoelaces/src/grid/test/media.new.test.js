import query from '../media.new';

describe('query', () => {
  const colSize = (num) => `${num * 100}%`;

  const colWidth = (size) => {
    if (!size) return {};
    if (size === 'auto') {
      return {
        flexGrow: 1,
        flexBasis: 0,
        maxWidth: '100%',
      };
    }

    const width = colSize(size);

    return {
      flexBasis: `${width}`,
      maxWidth: `${width}`,
    };
  };

  const props = {
    xs: 1,
    md: 1 / 2,
    lg: 1 / 3,
  };

  it('works', () => {
    expect(query(colWidth)(props)).toMatchSnapshot();
  });
});
