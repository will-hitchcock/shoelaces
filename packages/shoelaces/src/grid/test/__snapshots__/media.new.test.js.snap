// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query works 1`] = `
Object {
  "@media (min-width: 0)": Object {
    "flexBasis": "100%",
    "maxWidth": "100%",
  },
  "@media (min-width: 48em)": Object {
    "flexBasis": "50%",
    "maxWidth": "50%",
  },
  "@media (min-width: 62em)": Object {
    "flexBasis": "33.33333333333333%",
    "maxWidth": "33.33333333333333%",
  },
}
`;
